### PR TITLE
TOK-525: values-digits-display

### PR DESCRIPTION
**Pull request by franciscotobar**
[TOK-525](https://rsklabs.atlassian.net/browse/TOK-525)

### CODE REVIEW

File: src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx [hunk 1]
```diff
@@ -60,14 +60,14 @@ export const BuildersLeaderBoardTable: FC = () => {
       rewardPercentage: (a: IRewardData, b: IRewardData) =>
         Number(a.rewardPercentage.current - b.rewardPercentage.current),
       lastCycleRewards: (a: IRewardData, b: IRewardData) => {
-        const aValue = a.lastCycleReward.rif.crypto.value + a.lastCycleReward.rbtc.crypto.value
-        const bValue = b.lastCycleReward.rif.crypto.value + b.lastCycleReward.rbtc.crypto.value
-        return aValue - bValue
+        const aValue = a.lastCycleReward.rif.amount.value + a.lastCycleReward.rbtc.amount.value
+        const bValue = b.lastCycleReward.rif.amount.value + b.lastCycleReward.rbtc.amount.value
+        return Number(aValue - bValue)
```
Comment (antomor) [inline src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx:65]: Don't we risk to lose precision in this way?

File: src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx [hunk 2]
```diff
@@ -60,14 +60,14 @@ export const BuildersLeaderBoardTable: FC = () => {
...
       },
       estimatedRewards: (a: IRewardData, b: IRewardData) => {
-        const aValue = a.estimatedReward.rif.crypto.value + a.estimatedReward.rbtc.crypto.value
-        const bValue = b.estimatedReward.rif.crypto.value + b.estimatedReward.rbtc.crypto.value
-        return aValue - bValue
+        const aValue = a.estimatedReward.rif.amount.value + a.estimatedReward.rbtc.amount.value
+        const bValue = b.estimatedReward.rif.amount.value + b.estimatedReward.rbtc.amount.value
+        return Number(aValue - bValue)
```
Comment (antomor) [inline src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx:70]: Don't we risk to lose precision in this way?

File: src/app/collective-rewards/rewards/builders/hooks/useGetBuildersRewards.ts [hunk 3]
```diff
@@ -152,30 +151,30 @@ export const useGetBuildersRewards = ({ rif, rbtc }: { [token: string]: Token },
...
+      const estimatedRifAmount = (rewardsAmountRif * rewardPercentageToApply) / weiPerEther
 
       // calculate rbtc estimated rewards
       const rewardRbtc = rewardsCoinbase ?? 0n
       const rewardsAmountRbtc = totalPotentialRewards
         ? (rewardRbtc * builderRewardShares) / totalPotentialRewards
         : 0n
-      const estimatedRbtcInHuman =
-        Number(formatBalanceToHuman(rewardsAmountRbtc)) * (rewardPercentageToApply / 100)
+      const estimatedRbtcAmount = (rewardsAmountRbtc * rewardPercentageToApply) / weiPerEther
```
Comment (antomor) [inline src/app/collective-rewards/rewards/builders/hooks/useGetBuildersRewards.ts:170]: I'd create a function for these

File: src/app/collective-rewards/rewards/builders/EstimatedRewards.tsx [hunk 4]
```diff
@@ -75,10 +74,10 @@ const TokenRewards: FC<TokenRewardsProps> = ({ builder, gauge, token: { id, symb
   const rewardsAmount =
     rewardShares && totalPotentialRewards ? (rewards * rewardShares) / totalPotentialRewards : 0n
   // The complement of the reward percentage is applied to the estimated rewards since are from the builder's perspective
-  const estimatedRewardsInHuman =
-    Number(formatBalanceToHuman(rewardsAmount)) * (1 - rewardPercentageToApply / 100)
+  const weiPerEther = parseUnits('1', 18)
+  const estimatedRewards = (rewardsAmount * (weiPerEther - rewardPercentageToApply)) / weiPerEther
```
Comment (antomor) [inline src/app/collective-rewards/rewards/builders/EstimatedRewards.tsx:78]: We could use the previously defined function here

File: src/app/collective-rewards/rewards/hooks/useGetBackerRewardPercentage.ts [hunk 5]
```diff
@@ -23,7 +23,14 @@ export const useGetBackerRewardPercentage = (builder: Address, timestampInSecond
 
     const [previous, next, cooldownEndTime] = data
 
-    const percentageData = getPercentageData(previous, next, cooldownEndTime, timestampInSeconds)
+    const currentTimestamp = timestampInSeconds ?? Math.floor(Date.now() / 1000)
+    const current = currentTimestamp < cooldownEndTime ? previous : next
```
Comment (antomor) [inline src/app/collective-rewards/rewards/hooks/useGetBackerRewardPercentage.ts:27]: Why did we remove this function if we have the same code here and in the `useGetBackersRewardPercentage`?

File: src/app/collective-rewards/rewards/hooks/useGetBackersRewardPercentage.ts [hunk 6]
```diff
@@ -28,25 +28,31 @@ export const useGetBackersRewardPercentage = (builders: Address[], timestampInSe
...
 
-        const [current, next, cooldownEndTime] = (contractResults[index].result || [0n, 0n, 0n]) as [
+        const [previous, next, cooldownEndTime] = (contractResults[index].result || [0n, 0n, 0n]) as [
           bigint,
           bigint,
           bigint,
         ]
-        const result = getPercentageData(current, next, cooldownEndTime, timestampInSeconds)
-        acc[gauge] = result
+        const current = currentTimestamp < cooldownEndTime ? previous : next
```
Comment (antomor) [inline src/app/collective-rewards/rewards/hooks/useGetBackersRewardPercentage.ts:45]: we use the same logic in `useGetBackerRewardPercentage`

File: src/app/collective-rewards/rewards/types.ts [hunk 7]
```diff
@@ -30,3 +31,10 @@ export type RewardDetails = {
 }
 
 export type BuilderRewardDetails = RewardDetails & { gauge: Address }
+
+// TODO: this is a duplicate of BackerRewardPercentage
```
Comment (antomor) [inline src/app/collective-rewards/rewards/types.ts:35]: Let's remove this if it's a duplicate

File: src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx (hunk 1 above)
Comment (franciscotobar) [reply src/app/collective-rewards/leaderboard/BuildersLeaderBoardTable.tsx:65]: We may, let me change the bigint comparison.

File: src/app/collective-rewards/rewards/utils/formatMetrics.ts [hunk 8]
```diff
@@ -1,16 +1,71 @@
...
+    return '0'
+  }
+  const { decimals, displayDecimals } = symbols[symbol.toLocaleLowerCase()] ?? {
+    decimals: 18,
+    displayDecimals: 2,
+  }
+  const amount = Number(formatUnits(value, decimals))
+  const minimumAmount = 1 / Math.pow(10, decimals)
+
+  if (amount > 0 && amount < minimumAmount) {
```
[the remark this answers is not in the document]
Comment (antomor) [reply src/app/collective-rewards/rewards/utils/formatMetrics.ts]: @franciscotobar I think this is still valid.

File: src/app/collective-rewards/rewards/utils/formatter.ts [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  const amount = Number(formatUnits(value, decimals))
+  const minimumAmount = 1 / Math.pow(10, decimals)
+
+  if (amount > 0 && amount <= minimumAmount) {
+    return `<0${displayDecimals > 0 ? '.'.padEnd(displayDecimals, '0') + '1' : ''}`
+  }
+
+  if (amount > 0 && amount < 1 && displayDecimals === 0) {
+    return '<1'
+  }
```
Comment (antomor) [inline src/app/collective-rewards/rewards/utils/formatter.ts:73]: this part isn't very clear to me, I think we should format it a bit

**Review by antomor — APPROVED** [review]
LGTM

**Pull request merged**